src/redux/uniqueTokens.js

```diff
@@ -64,46 +64,56 @@ export const uniqueTokensRefreshState = () => async (dispatch, getState) => {
     return;
   }
 
-  dispatch(watchUniqueTokens());
+  dispatch(fetchUniqueTokens());
 };
 
 const fetchUniqueTokens = () => async (dispatch, getState) => {
   dispatch({ type: UNIQUE_TOKENS_GET_UNIQUE_TOKENS_REQUEST });
   const { accountAddress, network } = getState().settings;
   const { assets } = getState().data;
   const { uniqueTokens: existingUniqueTokens } = getState().uniqueTokens;
+  const shouldUpdateInBatches = isEmpty(existingUniqueTokens);
+
+  let shouldStopFetching = false;
+  let page = 0;
+  let uniqueTokens = [];
+
   try {
-    const uniqueTokens = await apiGetAccountUniqueTokens(accountAddress);
+    while (!shouldStopFetching) {
+      const newPageResults = await apiGetAccountUniqueTokens(
+        accountAddress,
+        page
+      );
+      uniqueTokens = concat(uniqueTokens, newPageResults);
+      shouldStopFetching =
+        newPageResults.length < UNIQUE_TOKENS_LIMIT_PER_PAGE ||
+        uniqueTokens.length >= UNIQUE_TOKENS_LIMIT_TOTAL;
+      page += 1;
+      if (shouldUpdateInBatches) {
+        dispatch({
+          payload: uniqueTokens,
+          type: UNIQUE_TOKENS_GET_UNIQUE_TOKENS_SUCCESS,
+        });
+      }
+    }
+
+    if (!shouldUpdateInBatches) {
+      dispatch({
+        payload: uniqueTokens,
+        type: UNIQUE_TOKENS_GET_UNIQUE_TOKENS_SUCCESS,
+      });
+    }
     const existingFamilies = getFamilies(existingUniqueTokens);
     const newFamilies = getFamilies(uniqueTokens);
     const incomingFamilies = without(newFamilies, ...existingFamilies);
     if (incomingFamilies.length) {
       const dedupedAssets = dedupeAssetsWithFamilies(assets, incomingFamilies);
       dispatch(dataUpdateAssets(dedupedAssets));
     }
+
     saveUniqueTokens(uniqueTokens, accountAddress, network);
-    dispatch({
-      payload: uniqueTokens,
-      type: UNIQUE_TOKENS_GET_UNIQUE_TOKENS_SUCCESS,
-    });
   } catch (error) {
     dispatch({ type: UNIQUE_TOKENS_GET_UNIQUE_TOKENS_FAILURE });
-    captureException(error);
```

Comment:
I'd still capture this. If opensea API starts failing often we would know.

Reply:
Sure sounds good